"Appear disabled while paused".

ErezVolk opened the pull request:
Thanks for a wonderful little app! Here's something I find useful. I like to be able to tell at a glance whether the timer is paused or not. I've added an option to gray out the time.

This is my first time ever writing anything for OSX, using Xcode, etc, so everything may be horribly wrong.

<img width="352" alt="bildschirmfoto 2017-12-23 um 12 34 54" src="https://user-images.githubusercontent.com/22130243/34318989-d08107de-e7dd-11e7-95be-a923b811173e.png">

Commit:
Added an option to gray out when paused

ErezVolk commented on ThymeAppDelegate.m:
```diff
@@ -214,17 +214,24 @@ - (IBAction)onAboutClick:(id)sender {
+    NSStatusBarButton *button = [statusItem button];
+    
     if ([self.stopwatch isStopped]) {
-        [statusItem setLength:26.0];
```
The docs said `setTitle` etc. were deprecated and you should now use the button. If this is breaking anything, it can go, of course.

smokris commented on PreferencesWindowController.xib:
```diff
@@ -1,687 +1,100 @@
+    <objects>
+        <customObject id="-2" userLabel="File's Owner" customClass="PreferencesWindowController">
+            <connections>
+                <outlet property="appearDisabledWhilePausedButton" destination="ljf-W9-TJb" id="KZj-J0-BBb"/>
```
```suggestion
                <outlet property="appearDisabledWhilePaused" destination="ljf-W9-TJb" id="KZj-J0-BBb"/>
```

ErezVolk replied:
Fair enough, but I think it's better to be consistent with the other checkbox names in the same dialog (`pauseOnSleepButton`, `pauseOnScreensaverButton`). Don't you agree?

smokris reviewed (changes requested):
.

smokris commented on PreferencesWindowController.h:
```diff
@@ -15,12 +15,14 @@
     SRRecorderControl *finishShortcutRecorder;
     NSButton *pauseOnSleepButton;
     NSButton *pauseOnScreensaverButton;
+    NSButton *appearDisabledWhilePaused;
```
```suggestion
    NSButton *appearDisabledWhilePausedButton;
```

smokris commented on PreferencesWindowController.h:
```diff
@@ -15,12 +15,14 @@
 @property (nonatomic, retain) IBOutlet SRRecorderControl *finishShortcutRecorder;
 @property (nonatomic, retain) IBOutlet NSButton *pauseOnSleepButton;
 @property (nonatomic, retain) IBOutlet NSButton *pauseOnScreensaverButton;
+@property (nonatomic, retain) IBOutlet NSButton *appearDisabledWhilePaused;
```
```suggestion
@property (nonatomic, retain) IBOutlet NSButton *appearDisabledWhilePausedButton;
```

smokris commented on PreferencesWindowController.m:
```diff
@@ -19,6 +19,7 @@ @implementation PreferencesWindowController
 @synthesize finishShortcutRecorder;
 @synthesize pauseOnSleepButton;
 @synthesize pauseOnScreensaverButton;
+@synthesize appearDisabledWhilePaused;
```
```suggestion
@synthesize appearDisabledWhilePausedButton;
```

smokris commented on PreferencesWindowController.m:
```diff
@@ -42,6 +43,7 @@ - (void)windowDidLoad
     
     [self.pauseOnSleepButton bind:NSValueBinding toObject:defaults withKeyPath:@"values.pauseOnSleep" options:nil];
     [self.pauseOnScreensaverButton bind:NSValueBinding toObject:defaults withKeyPath:@"values.pauseOnScreensaver" options:nil];
+    [self.appearDisabledWhilePaused bind:NSValueBinding toObject:defaults withKeyPath:@"values.appearDisabledWhilePaused" options:nil];
```
```suggestion
    [self.appearDisabledWhilePausedButton bind:NSValueBinding toObject:defaults withKeyPath:@"values.appearDisabledWhilePaused" options:nil];
```

ErezVolk replied:
Oh, I see now! That's really weird, I can't see how I missed it. Thank you!

smokris reviewed (changes requested):
Oh, I should've explained — when I built your PR and ran it and opened the Preferences window, I got an error message saying that it couldn't connect `appearDisabledWhilePausedButton` to its outlet in the class `PreferencesWindowController` (whose property name lacks the `Button` suffix).

Going the other direction (adding the `Button` suffix to the property) sounds good to me.  I've added those change suggestions.

ErezVolk committed:
Update PreferencesWindowController.h

Fix by @smokris for `Failed to connect (appearDisabledWhilePausedButton) outlet from (PreferencesWindowController) to (NSButton)`

Co-Authored-By: Steve Mokris <smokris@softpixel.com>

ErezVolk committed:
Update PreferencesWindowController.h

Fix by @smokris for `Failed to connect (appearDisabledWhilePausedButton) outlet from (PreferencesWindowController) to (NSButton)`

Co-Authored-By: Steve Mokris <smokris@softpixel.com>

ErezVolk committed:
Update PreferencesWindowController.m

Fix by @smokris for `Failed to connect (appearDisabledWhilePausedButton) outlet from (PreferencesWindowController) to (NSButton)`

Co-Authored-By: Steve Mokris <smokris@softpixel.com>

ErezVolk committed:
Update PreferencesWindowController.m

Fix by @smokris for `Failed to connect (appearDisabledWhilePausedButton) outlet from (PreferencesWindowController) to (NSButton)`

Co-Authored-By: Steve Mokris <smokris@softpixel.com>

ErezVolk committed:
Recreated button

ErezVolk committed:
Merge remote-tracking branch 'upstream/master' into appear-disabled-while-paused

ErezVolk committed:
Updated DDHotKey usage

ErezVolk commented:
Update: I upgraded DDHotKey and fixed its usage (looks like you're not supposed to create your own `DDHotKeyCenter` anymore, and use `[DDHotKeyCenter sharedHotKeyCenter]` instead).
